lib/spack/spack/solver/asp.py

```diff
@@ -409,15 +435,32 @@ def specs(self):
         """List of concretized specs satisfying the initial
         abstract request.
         """
-        # The specs were already computed, return them
-        if self._concrete_specs:
-            return self._concrete_specs
+        if not self._concrete_specs:
+            self._compute_specs_from_answer_set()
+        return self._concrete_specs
 
-        # Assert prerequisite
-        msg = 'cannot compute specs ["satisfiable" is not True ]'
-        assert self.satisfiable, msg
+    @property
+    def unsolved_specs(self):
+        """List of abstract input specs that were not solved."""
+        if not self._unsolved_specs:
+            self._compute_specs_from_answer_set()
+        return self._unsolved_specs
+
+    @property
+    def specs_by_input(self):
+        if not self._concrete_specs_by_input:
```

Comment:
I don't think the same applies here because the only way to have an empty list is if it's unsat, but we might as well just in case.
```suggestion
        if self._concrete_specs_by_input is None:
```

Reply:
Fixed in 03e27c2